### PR TITLE
Fix mac aarch compilation

### LINKED ISSUE #1302: 在 M1/M2 Chip 的 Mac 上无法编译通过
详细错误如下：

```
[ERROR] Failed to execute goal on project sofa-rpc-remoting-http: Could not resolve dependencies for project com.alipay.sofa:sofa-rpc-remoting-http:jar:5.9.1: io.netty:netty-tcnative-boringssl-static:jar:osx-aarch_64:2.0.25.Final was not found in https://repo.maven.apache.org/maven2 during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of central has elapsed or updates are forced -> [Help 1]
```

需要升级 netty-tcnative-boringssl-static 的版本。

### PR DESCRIPTION
**Pull request by khotyn**
Fix #1302 

Dependency upgrade to compatible with aarch and IPv6:

* bolt: 1.5.10 --> 1.6.6
* sofa registry: 5.2.0 --> 6.3.0
* grpc: 1.33.1 --> 1.53.0
* protobuf: 3.16.3 --> 3.22.0
* netty-tcnative-boringssl-static: 2.0.25.Final --> 2.0.59.Final
* protoc: 3.11.0 --> 3.22.0
* nacos-embedded: 0.2.0 --> 0.3.0

FIx test cases:

* NacosRegistryTest(L222): Since all the provider are unregistered, so the provider list should be 0(It seems there is a bug before).
* BaseNacosTest: use newer nacos version that is compatible with aarch.
* Add a sofa common profile StringUtil since in newer sofa common profile package StringUtil is deleted, but sofa registry need it.
* TripleServerTest: Assert status code rather than assert exception message to make the test case more stable.

### CODE REVIEW

**Comment by khotyn** [comment]
可能还需要考虑下 gRPC 的版本的变化可能会导致的兼容性问题。

**Comment by JervyShi** [comment]
看起来 Nacos 还没升级支持 m1/m2 的 jraft 版本 https://github.com/alibaba/nacos/issues/9015

**Comment by khotyn** [comment]
@JervyShi 发布版本还没有，master 的代码已经升级了。

**Comment by JervyShi** [comment]
https://github.com/alibaba/nacos/pull/9790 看到有人已经提交了代码，等 Nacos 发版了我发个 [nacos-embedded](https://github.com/jervyshi/nacos-embedded) 版本

File: example/pom.xml [hunk 1]
```diff
@@ -194,7 +194,7 @@
                 <artifactId>protobuf-maven-plugin</artifactId>
                 <version>0.5.1</version>
                 <configuration>
-                    <protocArtifact>com.google.protobuf:protoc:3.7.1:exe:${os.detected.classifier}</protocArtifact>
+                    <protocArtifact>com.google.protobuf:protoc:${protoc.version}:exe:${os.detected.classifier}</protocArtifact>
                     <pluginId>grpc-java</pluginId>
                     <pluginArtifact>io.grpc:protoc-gen-grpc-java:${grpc.version}:exe:${os.detected.classifier}</pluginArtifact>
```
Comment (OrezzerO) [inline example/pom.xml:199]: 在 M1上编译 Dubbo 的时候发现 , pluginArtifact 中${os.detected.classifier} 需要改成 `osx-x86_64`,要特殊处理下, 这里应该也一样.

**Comment by JervyShi** [comment]
> [alibaba/nacos#9790](https://github.com/alibaba/nacos/pull/9790) 看到有人已经提交了代码，等 Nacos 发版了我发个 [nacos-embedded](https://github.com/jervyshi/nacos-embedded) 版本

https://github.com/JervyShi/nacos-embedded/releases/tag/v0.3.0 已经 Release，Maven 中央仓库已上传，但需要一些时间才能查到

**Comment by khotyn** [comment]
@JervyShi Thank you~~

**Comment by khotyn** [comment]
Still waiting https://github.com/sofastack/sofa-registry/issues/327 to complete so we can have a sofa registry-test version that is compatible with aarch.

**Comment by khotyn** [comment]
All the test case passes! @OrezzerO @EvenLjj @Lo1nt Pls help review~~~, thanks

File: registry/registry-sofa/pom.xml [hunk 2]
```diff
@@ -66,21 +66,19 @@
...
         </dependency>
         <dependency>
             <groupId>junit</groupId>
             <artifactId>junit</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>io.prometheus</groupId>
+            <artifactId>simpleclient_hotspot</artifactId>
+            <version>0.16.0</version>
```
Comment (EvenLjj) [inline registry/registry-sofa/pom.xml:79]: The version management of this jar package can be moved to bom.xml.

File: registry/registry-sofa/src/test/java/com/alipay/sofa/common/profile/StringUtil.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alipay.sofa.common.profile;
+
+/**
+ * @author khotyn
+ * @version StringUtil.java, v 0.1 2023年07月05日 14:40 khotyn
+ */
+public class StringUtil {
```
Comment (EvenLjj) [inline registry/registry-sofa/src/test/java/com/alipay/sofa/common/profile/StringUtil.java:23]: Haven't seen where use of the `com. alipay. sofa. common. profile. StringUtil` class in test.

Comment (khotyn) [reply registry/registry-sofa/src/test/java/com/alipay/sofa/common/profile/StringUtil.java:23]: sofa-registry-test depends on this class.

File: all/pom.xml [hunk 4]
```diff
@@ -61,16 +61,16 @@
...
         <lookout.version>1.4.1</lookout.version>
         <bytebuddy.version>1.9.8</bytebuddy.version>
-        <sofa.registry.version>5.2.0</sofa.registry.version>
+        <sofa.registry.version>6.3.0</sofa.registry.version>
         <bolt.swagger.version>1.6.9</bolt.swagger.version>
         <asm.version>7.0</asm.version>
         <httpclient.version>4.5.13</httpclient.version>
         <httpcore.version>4.4.13</httpcore.version>
-        <grpc.version>1.33.1</grpc.version>
+        <grpc.version>1.53.0</grpc.version>
```
[the remark this answers is not in the document]
Comment (khotyn) [reply all/pom.xml:73]: @EvenLjj We can verify it publicly, 😄

File: registry/registry-sofa/src/test/java/com/alipay/sofa/common/profile/StringUtil.java (hunk 3 above)
Comment (khotyn) [reply registry/registry-sofa/src/test/java/com/alipay/sofa/common/profile/StringUtil.java:23]: Related SOFARegistry issue: https://github.com/sofastack/sofa-registry/pull/332.

After SOFARegistry new version released, we can delete this class.

File: example/pom.xml (hunk 1 above)
Comment (Lo1nt) [reply example/pom.xml:199]: 看了下${os.detected.classifier}在 m1 下会被解析为 osx-aarch_64 , 老版本下会找不到，pr这里升级到3.22.0已经解决这个问题了。 (https://repo1.maven.org/maven2/com/google/protobuf/protoc/3.22.0/protoc-3.22.0-osx-aarch_64.exe)

**Review by Lo1nt — APPROVED** [review]

**Review by EvenLjj — APPROVED** [review]
LGTM

**Pull request merged**